schema_utils.py

```diff
@@ -53,6 +53,25 @@
 SCHEMA_TYPE_LIST = 'list'
 SCHEMA_TYPE_UNICODE = 'unicode'
 
+OPERATOR_PRECEDENCE = {
+    '+': 1,
+    '-': 1, # Binary subtraction.
+    '~': 1, # Unary negation.
```

Comment:
Hm. You're expecting students to type '~'?

Reply:
No, not at all. While parsing the string (in the tokenize function [here](https://github.com/oppia/oppia/pull/9336/files#diff-781314b975a52120938e342ae5ffd787R302)) I've checked if the '-' sign represents unary negation or binary subtraction and replaced '-' with '~' if the former is true. This is done so that while validating the postfix expression, I would know if the '-' is expecting two operators or one. So basically, `a - (-b)` would be changed to `a - (~b)`.

Reply:
I'd added a comment explaining it [here](https://github.com/oppia/oppia/pull/9336/files#diff-781314b975a52120938e342ae5ffd787R317).

Reply:
No, I mean, adding a comment here explaining the point above (i.e. something along the lines of "you do not expect students to type that character and that it's being added by us as part of the XXX function below").

Think about a future dev who is, perhaps, trying to add a new operator to this OPERATOR_PRECEDENCE constant. They won't know that you have a comment somewhere else in the file relating to this.

Reply:
This conversation is obsolete due to the new changes. Now, there is no need for replacing - with ~.